kornia/core/tensor_wrapper.py

```diff
@@ -0,0 +1,146 @@
+
+
+class TensorWrapper:
+    def __init__(self, data: Tensor) -> None:
+        self.__dict__["_data"] = data
+        self.__dict__["used_attrs"] = set()
+        self.__dict__["used_calls"] = set()
+
+    @property
+    def data(self) -> Tensor:
+        return self._data
+
+    def __repr__(self) -> str:
+        return f"{self.data}"
+
+    def __getattr__(self, name):
+        if name in self.__dict__:
+            return self.__dict__[name]
+        self.used_attrs.add(name)
+
+        val = getattr(self._data, name)
+
+        # NOTE: not clear is really needed
+        # If it's a method
+        # if callable(val):
+        #    c = getattr(type(self._data), name)
+        #    # Don't append self to args if classmethod/staticmethod
+        #    if c is val:
+        #        return lambda *a, **kw: wrap(self.__torch_function__(c, (type(self),), args=a, kwargs=kw), type(self))
+        #    # Otherwise append self to args
+        #    return lambda *a, **kw: wrap(
+        #        #self.__torch_function__(c, (type(self),), args=(self,) + a, kwargs=kw), type(self)
+        #    )
+
+        return wrap(val, type(self))
+
+    def __setattr__(self, name, value):
+        if name in self.__dict__:
+            self.__dict__[name] = value
+
+        self.used_attrs.add(name)
+        setattr(self._data, name, value)
+
+    def __setitem__(self, key, value):
+        self._data[key] = value
+
+    def __getitem__(self, key):
+        return wrap(self._data[key], type(self))
+
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+        # Find an instance of this class in the arguments
+        args_of_this_cls = []
+        for a in args:
+            if isinstance(a, cls):
+                args_of_this_cls.append(a)
+            elif isinstance(a, collections.abc.Sequence):
+                args_of_this_cls.extend(el for el in a if isinstance(el, cls))
+        # assert len(args_of_this_cls) > 0
+        for a in args_of_this_cls:
+            a.used_calls.add(func)
+        args = unwrap(tuple(args))
+        kwargs = {k: unwrap(v) for k, v in kwargs.items()}
+
+        return wrap(func(*args, **kwargs), cls)
+
+    # TODO: `def __add__(self, other) -> Self:` when mypy release >0.991
+    def __add__(self, other):
+        return self.__unary_op__(torch.add, other)
+
+    def __mul__(self, other):
+        return self.__unary_op__(torch.mul, other)
+
+    def __sub__(self, other):
+        return self.__unary_op__(torch.sub, other)
+
+    def __truediv__(self, other):
+        return self.__unary_op__(torch.true_divide, other)
+
+    def __floordiv__(self, other):
+        return self.__unary_op__(torch.floor_divide, other)
+
+    def __ge__(self, other):
+        return self.__unary_op__(torch.ge, other)
+
+    def __gt__(self, other):
+        return self.__unary_op__(torch.gt, other)
+
+    def __lt__(self, other):
```

Comment:
@johnnv1 something to investigate is why codecov doesn't recognize tests here

Reply:
i think it's because typing -- we stay uploading typing coverage, so codecov gets sometimes lost

Reply:
I plan to soon enable some flags in mypy to guarantee typing for everything. Then we can get rid of typing coverage without worrying (but I still don't think anyone uses codecov typing coverage anyway 😅)

Reply:
yeah, sounds good. I support to remove codecov typing coverage once stable